src/main/java/com/epam/java/training/dao/ProductDAO.java

```diff
@@ -0,0 +1,48 @@
+package com.epam.java.training.dao;
+
+import java.util.List;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.epam.java.training.entity.Product;
+@Transactional
```

Comment:
usually we keep @Transactional annotation at service layer. Service layer may call different DAO to perform DB operations